### PR TITLE
Supporting single upload cancellation

### PR DESCRIPTION
**Pull request by ola-rozenfeld**

### CODE REVIEW

File: go/pkg/client/cas.go [hunk 1]
```diff
@@ -91,10 +93,40 @@ func (c *Client) uploadProcessor() {
...
+			for _, r := range buffer {
+				if r.c != req.c || r.wait != req.wait {
+					newBuffer = append(newBuffer, r)
+				}
+			}
+			buffer = newBuffer
+			st, ok := c.casUploads[req.c.Digest()]
+			if ok {
+				st.mu.Lock()
+				var newClients []chan<- *uploadResponse
```
Comment (ramymedhat) [inline go/pkg/client/cas.go:115]: Can you add a comment clarifying newClients? Something along the lines of it being a list of the remaining clients after the current cancellation requestor drops out?

File: go/pkg/client/cas.go [hunk 2]
```diff
@@ -58,6 +59,7 @@ type uploadState struct {
 	// exactly once for a given state object (this is the whole point of the algorithm).
 	mu      sync.Mutex
 	clients []chan<- *uploadResponse
+	cancel  func()
```
Comment (ramymedhat) [inline go/pkg/client/cas.go:62]: Can you update the comment above to indicate that mu now also protects the cancel function?

**Review by ramymedhat — APPROVED** [review]

File: go/pkg/client/cas.go (hunk 2 above)
Comment (ola-rozenfeld) [reply go/pkg/client/cas.go:61]: Done.

File: go/pkg/client/cas.go (hunk 1 above)
Comment (ola-rozenfeld) [reply go/pkg/client/cas.go]: Did one better and renamed the variable so it's clear.

**Pull request merged**